lib/Config.js

```diff
@@ -67,13 +86,76 @@ class Config {
             });
         }
 
-        assert(typeof config.regions === 'object',
-               'bad config: the list of regions is mandatory');
-        assert(Object.keys(config.regions).every(
-               r => typeof r === 'string' && config.regions[r] instanceof Array
-               && config.regions[r].every(e => typeof e === 'string')),
-               'bad config: regions must be a set of {region: [endpoints]}');
-        this.regions = config.regions;
+        // legacy
+        if (config.regions !== undefined) {
+            assert(typeof config.regions === 'object',
+                'bad config: the list of regions is mandatory');
+            assert(Object.keys(config.regions).every(
+                r => typeof r === 'string' && config.regions[r] instanceof Array
+                && config.regions[r].every(e => typeof e === 'string')),
+                'bad config: regions must be a set of {region: [endpoints]}');
+            this.regions = config.regions;
+        }
+
+        this.locationConstraints = {};
+        if (config.locationConstraints !== undefined) {
+            assert(typeof config.locationConstraints === 'object',
+                'bad config: locationConstraints must be an object');
+            Object.keys(config.locationConstraints).forEach(l => {
+                assert(typeof config.locationConstraints[l].type === 'string' &&
```

Comment:
Can this be simplified to `typeof l.type === 'string'`? (And throughout these checks.)